src/tsolvers/arraysolver/ArraySolver.cc

```diff
@@ -144,7 +144,8 @@ void ArraySolver::declareAtom(PTRef tr) {
                 if (logic.isArrayStore(term)) {
                     storeTerms.insert(eref);
```

Comment:
I assume that `logic.isArrayStore(term) && logic.isArraySelect(term)` is impossible. Hence we can use `return` here?

Reply:
Maybe also an assert such as `logic.isArrayStore(term) => logic.isArraySort(logic.getSortRef(term))` at the beginning could be useful.

Reply:
Added an assertion at the beginning and the return in the nested `if`.